Troubleshooting guides cleanup (#7910)
kyma-project/kyma: docs/service-mesh/10-03-connection-refused.md

---
title: Connection refused errors
type: Troubleshooting
---

Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar with a valid TLS certificate to allow communication. Connection between a service with a sidecar and a service without a sidecar results in a `Connection reset by peer` or a `GOAWAY` response. 

- To enable sidecar injection for Pods of existing Services, restart them after upgrading to Kyma 1.0 or higher.

> - If you don't want to use a sidecar, you can whitelist a Service and disable TLS traffic for it by creating a [DestinationRule](https://istio.io/docs/reference/config/networking/destination-rule/).  

Reviewer: To tell all the clients(other sidecars) to do not use mTLS when connecting to my service we use DestinationRule. To accept PERMISSIVE connections for my service that has a sidecar we use Policy - at least for now as there are now in 1.5 new CRs for that, but this is out of scope as we don't have it yet. To disable sidecar injection we follow step bellow :)

---
title: Connection refused errors
type: Troubleshooting
---

> Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar with a valid TLS certificate to allow communication. Connection between a service with a sidecar and a service without a sidecar results in a `Connection reset by peer` or a `GOAWAY` response. 

Reviewer 2: ```suggestion
Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar with a valid TLS certificate to allow communication. Attempts to establish connection between a service with a sidecar and a service without a sidecar results in a `Connection reset by peer` or a `GOAWAY` response. 
```

> Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar with a valid TLS certificate to allow communication. Attempts to establish connection between a service with a sidecar and a service without a sidecar results in a `Connection reset by peer` or a `GOAWAY` response. 

Reviewer 2: ```suggestion
Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar with a valid TLS certificate to allow communication. Attempts to establish connection between a service with a sidecar and a service without a sidecar result in a `Connection reset by peer` or a `GOAWAY` response. 
```

Sorry, I missed that. Attempts -> result (plural)

- To enable sidecar injection for Pods of existing Services, restart them after upgrading to Kyma 1.0 or higher.

> - To allow connections to a Service with a sidecar from Services without a sidecar, create a [Policy](https://istio.io/docs/reference/config/security/istio.authentication.v1alpha1/) in a `PERMISSIVE` mode.

Reviewer 2: ```suggestion
- To allow connections between a Service with a sidecar and Services without a sidecar, create an [Authenitcation Policy](https://istio.io/docs/reference/config/security/istio.authentication.v1alpha1/) in the `PERMISSIVE` mode.
```

Author: The direction is important in this case.
